controllers/volumereplicationgroup_controller.go

```diff
@@ -351,6 +364,79 @@ func (v *VRGInstance) validateVRGState() error {
 	return nil
 }
 
+func (v *VRGInstance) restorePVs() error {
+	vrgCondition := findCondition(v.instance.Status.Conditions, VRGConditionAvailable)
```

Comment:
For testing purposes this is fine. We should adapt to something like so,
- For each PV
    - Check exists with an annotation (say ramenrestore: true)
    - If exists without annotation, conflict, error out
    - If exists with annotation, continue to next PV
    - If not exists, create with annotation

Towards the end of DeleteVRG, we can then garbage collect any PVs with the annotation that the application failed to create PVCs for and hence will never be claimed or garbage collected by the k8s controllers and CSI.

Reply:
I think I understand the idea.

Reply:
We will test this first and then implement? So that we know this scheme overall works?